Mathlib/GroupTheory/GroupAction/Blocks.lean

```diff
@@ -0,0 +1,490 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.Data.Setoid.Partition
+import Mathlib.GroupTheory.GroupAction.Basic
+import Mathlib.GroupTheory.GroupAction.Pointwise
+import Mathlib.GroupTheory.GroupAction.SubMulAction
+import Mathlib.GroupTheory.Subgroup.Actions
+
+/-! # Blocks
+
+Given `SMul G X`, an action of a type `G` on a type `X`, we define
+
+- the predicate `IsBlock G B` states that `B : Set X` is a block,
+  which means that the sets `g • B`, for `g ∈ G` form a partition of `X`.
```

Comment:
To me, `partition` implies that the translates cover all of `X`. But you don't require that, do you?

Reply:
No, the docstring was incorrect